IpfsImage.js

```diff
@@ -0,0 +1,267 @@
+/**
+ * Copyright (c) 2015-present, Facebook, Inc.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ * @providesModule Image
+ * @flow
+ * @format
+ */
+'use strict';
+
+const EdgeInsetsPropType = require('EdgeInsetsPropType');
+const ImageResizeMode = require('ImageResizeMode');
+const ImageSourcePropType = require('ImageSourcePropType');
+const ImageStylePropTypes = require('ImageStylePropTypes');
+const NativeMethodsMixin = require('NativeMethodsMixin');
+const NativeModules = require('NativeModules');
+const React = require('React');
+const PropTypes = require('prop-types');
+const ReactNativeViewAttributes = require('ReactNativeViewAttributes');
+const StyleSheet = require('StyleSheet');
+const StyleSheetPropType = require('StyleSheetPropType');
+
+const createReactClass = require('create-react-class');
+const flattenStyle = require('flattenStyle');
+const requireNativeComponent = require('requireNativeComponent');
+const resolveAssetSource = require('resolveAssetSource');
+
+const IpfsImageViewManager = NativeModules.IpfsImageViewManager;
+
+/**
+ * A React component for displaying different types of images,
+ * including network images, static resources, temporary local images, and
+ * images from local disk, such as the camera roll.
+ *
+ * See https://facebook.github.io/react-native/docs/image.html
+ */
+const IpfsImage = createReactClass({
```

Comment:
This is the JS layer of the new `IpfsImage`. This is actually the iOS version, still need to make the android version (RN uses different JS implementations of `Image` depending on platform).